docs/sdk/access-control/other-chains/stellar-access-control.md

```diff
@@ -0,0 +1,396 @@
+---
+sidebar_position: 6
+---
+
+# Lit Action Conditions: Reading State from Stellar
+
+Currently Stellar is not one of the [listed supported chains](../../../../versioned_docs/version-2.0/resources/supportedChains.md) by Lit. This guide provides an example of how you can use a Lit Action to make a call to Stellar's network to determine whether or not a user is allowed to decrypt some data encrypted using Lit.
+
+The source code for this guide can be found [here](https://github.com/LIT-Protocol/developer-guides-code/tree/master/lit-actions-conditions-reading-state-from-stellar).
+
+There are two parts to this guide:
+
+- Using Lit to provide decryption access control by reading state from Stellar
+- Creating the authorizing Stellar smart contract
+
+## Using Lit to Provide Decryption Access Control
+
+This guide will utilize the Lit SDK to encrypt some data, and a Lit Action with a secret number to determine whether or not authorization should be granted to decrypt the data using the Lit network.
+
+### Encrypting and Decrypting
+
+[This file](https://github.com/LIT-Protocol/developer-guides-code/blob/master/lit-actions-conditions-reading-state-from-stellar/lit/src/index.js) contains the Lit SDK code responsible for encrypting a string and calling the Lit Action to attempt decryption.
+
+#### Connecting to the Lit Habanero Network
+
+The first thing we need to do is create a Lit Client connected to the `habanero` network:
+
+```javascript
+const client = new LitJsSdk.LitNodeClientNodeJs({
+  litNetwork: "habanero",
+});
+await client.connect();
+```
+
+#### Creating an Auth Signature
+
+Then we need to create an [Auth Sig](../../../../versioned_docs/version-2.0/sdk/explanation/authentication/authsig):
+
+```javascript
+const authSig = await getAuthSig(client);
+```
+
+We use a provided private key (given to us as an environment variable) to create an `ethers` wallet, create a Sign in With Ethereum (SIWE) message, sign it, then return an object that is our Auth Sig.
+
+```javascript
+function getPrivateKey() {
+  if (process.env.PRIVATE_KEY === undefined)
+    throw new Error("Please provide the env: PRIVATE_KEY");
+  return process.env.PRIVATE_KEY;
+}
+
+function getWallet() {
+  return new ethers.Wallet(getPrivateKey());
+}
+
+async function getAuthSig(client) {
+  const wallet = getWallet();
+  const address = ethers.getAddress(await wallet.getAddress());
+  const messageToSign = (
+    await getSiweMessage(client, address)
+  ).prepareMessage();
+  const signature = await wallet.signMessage(messageToSign);
+
+  return {
+    sig: signature,
+    derivedVia: "web3.eth.personal.sign",
+    signedMessage: messageToSign,
+    address,
+  };
+}
+```
+
+Our method to create the SIWE message to sign looks like:
+
+```javascript
+async function getSiweMessage(client, address) {
+  const domain = "localhost";
+  const origin = "https://localhost/login";
+  const statement =
+    "This is a test statement.  You can put anything you want here.";
+
```

Comment:
Can we add something to this guide that says that the user should configure these to there web application context? These should be configured by the DApp developer.